packages/domains-table/src/domains-table/domains-table-email-indicator.tsx

```diff
@@ -1,36 +1,86 @@
-import { LoadingPlaceholder } from '@automattic/components';
 import { PartialDomainData } from '@automattic/data-stores';
-import { useI18n } from '@wordpress/react-i18n';
-//eslint-disable-next-line no-restricted-imports
-import { useGetMailboxes } from 'calypso/data/emails/use-get-mailboxes';
+import { translate } from 'i18n-calypso';
+import { getDomainType } from '../utils/get-domain-type';
 
 export const DomainsTableEmailIndicator = ( {
 	domain,
 	siteSlug,
 }: {
 	domain: PartialDomainData;
-	siteSlug?: string;
+	siteSlug: string;
 } ) => {
-	const { __ } = useI18n();
-	const { data: mailboxes = [], isLoading } = useGetMailboxes( domain.blog_id, {
-		retry: 2,
-	} );
+	// mailbox logic below moved from client/my-sites/domains/domain-management/list/domain-row.jsx
 
-	if ( isLoading || ! siteSlug ) {
-		return <LoadingPlaceholder style={ { width: '50%' } } />;
+	if ( ! [ 'mapped', 'registered' ].includes( getDomainType( domain ) ) ) {
+		return null;
 	}
 
-	if ( mailboxes.length === 0 ) {
+	const googleStatus = domain.google_apps_subscription?.status || '';
+	const titanStatus = domain.titan_mail_subscription?.status || '';
+
+	let message = null;
+
+	if ( googleStatus && ! [ '', 'no_subscription', 'other_provider' ].includes( googleStatus ) ) {
+		const count = domain?.google_apps_subscription?.total_user_count ?? 0;
+
+		message = translate( '%(count)d mailbox', '%(count)d mailboxes', {
+			count,
+			args: {
+				count,
+			},
+			comment: 'The number of GSuite mailboxes active for the current domain',
+		} );
+	} else if ( titanStatus && ( titanStatus === 'active' || titanStatus === 'suspended' ) ) {
+		const count = domain.titan_mail_subscription?.maximum_mailbox_count ?? 1;
+		message = translate( '%(count)d mailbox', '%(count)d mailboxes', {
+			args: {
+				count,
+			},
+			count,
+			comment: '%(count)d is the number of mailboxes for the current domain',
+		} );
+	} else if ( domain.email_forwards_count > 0 ) {
+		message = translate( '%(count)d forward', '%(count)d forwards', {
+			count: domain.email_forwards_count,
+			args: {
+				count: domain.email_forwards_count,
+			},
+			comment: 'The number of email forwards active for the current domain',
+		} );
+	}
+
+	if ( message ) {
 		return (
-			<a className="add-email-button" href={ `/email/${ domain.domain }/manage/${ siteSlug }` }>
-				+ { __( 'Add email ' ) }
+			<a
+				className="domains-table-view-email-button"
+				href={ emailManagementEdit( siteSlug, domain.domain ) }
+			>
+				{ message }
 			</a>
 		);
 	}
 
+	if ( ! domain.current_user_can_add_email ) {
+		return null;
+	}
+
 	return (
-		<>
-			{ __( 'Email ' ) } { mailboxes.length }
-		</>
+		<a
+			className="domains-table-add-email-button"
+			href={ `/email/${ domain.domain }/manage/${ siteSlug }` }
+		>
+			{ translate( '+ Add email' ) }
+		</a>
 	);
 };
+
+export function emailManagementEdit( siteSlug: string, domainName: string ) {
```

Comment:
I think this method should go in the `utils/paths.ts` file with other similar methods